edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentCommand.cs

```diff
@@ -135,52 +135,48 @@ async Task PushEdgeDeployment(CancellationToken token)
                     module =>
                     {
                         var combinedConfig = this.configProvider.GetCombinedConfig(module, this.runtimeInfo);
-                        // TODO: this is a workaround in preview to keep Agent from updating
                         var image = combinedConfig.Image;
-                        if (module.Name == Core.Constants.EdgeAgentModuleName &&
-                            this.currentmodules.Modules.TryGetValue(Core.Constants.EdgeAgentModuleName, out IModule edgeAgentCurrentModule))
+
+                        // TODO: this is a workaround in preview to keep Edge Agent from updating itself
+                        if (module.Name == Core.Constants.EdgeAgentModuleName)
                         {
-                            var currentAgent = this.configProvider.GetCombinedConfig(edgeAgentCurrentModule, this.runtimeInfo);
-                            image = currentAgent.Image;
+                            if(this.activeDeployment.HasValue)
+                            {
+                                var currentAgent = this.activeDeployment.OrDefault().Spec.First(agentModule => agentModule.Name == Core.Constants.EdgeAgentModuleName);
+                                image = currentAgent.Config.Image;
+                            }
+                            else
+                            {
+                                try
+                                {
+                                    // When CRD has not been created, use helm chart deployment details
+                                    var agentDeployment = this.client.ReadNamespacedDeployment(Core.Constants.EdgeAgentModuleName.ToLower(), this.deviceNamespace);
+                                    image = agentDeployment.Spec.Template.Spec.Containers.First(container => container.Name == Core.Constants.EdgeAgentModuleName.ToLower()).Image;
```

Comment:
The agent is running because the deployment exists. The catch block handles REST API failures (please see the UT for this scenario). In the very exceptional case of null dereference, the same exception logic applies.